### PR TITLE
Fix warnings and update ReadMe

### PR DESCRIPTION
**Pull request by aggarg**
Description
-----------
The demo `FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo` builds clean without warning. Also, make the demo output comprehensible.

Test Steps
-----------
Tested locally.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sirnish — APPROVED** [review]

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/ReadMe.md [hunk 1]
```diff
@@ -1,192 +1,199 @@
...
+   repo on the host where you want to run the echo server:
+   ```
+   git clone https://github.com/FreeRTOS/FreeRTOS-Libraries-Integration-Tests.git
+   ```
+1. Run the TCP echo server available at [tools/echo_server](https://github.com/FreeRTOS/FreeRTOS-Libraries-Integration-Tests/tree/main/tools/echo_server)
+   by running the following command:
+   ```
+   go run echo_server.go
+   ```
+1. Set `configECHO_SERVER_ADDR_STRING` in the [FreeRTOSConfig.h](FreeRTOSConfig.h)
```
Comment (tony-josi-aws) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/ReadMe.md:55]: Indexes of numbered lists seems to be 1 for all lines.

**Review by tony-josi-aws — APPROVED** [review]

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/ReadMe.md [hunk 2]
```diff
@@ -1,192 +1,199 @@
...
-2. `configNET_MASK0/3`        : Setup with the appropriate network mask.
-3. `configGATEWAY_ADDR0/3 `   : Setup with Default Gateway address of Client IP.
-4. `configDNS_SERVER_ADDR0/3` : Setup wit DNS Server address of Client IP.
-
-*Server Configuration :*
-1. `configECHO_SERVER_ADDR_STRING` as an IPv6 Server IP address for IPv6
-    validation, it can be updated to IPv4 address for IPv4 validations.
-2. `configECHO_SERVER_PORT` Needs to be setup for the Server port number.
-
-Note that, as delivered, configUSE_DHCPv6 is set to 0, so a static IP address is used.
```
Comment (moninom1) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/ReadMe.md:54]: Can we add this note somewhere?

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/FreeRTOSIPConfig.h [hunk 3]
```diff
@@ -44,7 +44,7 @@ extern void vLoggingPrintf( const char * pcFormatString,
 /* Set to 1 to print out debug messages.  If ipconfigHAS_DEBUG_PRINTF is set to
  * 1 then FreeRTOS_debug_printf should be defined to the function used to print
  * out the debugging messages. */
-#define ipconfigHAS_DEBUG_PRINTF    1
```
Comment (moninom1) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/FreeRTOSIPConfig.h:47]: Can we also add a note in readme to enabling these flags for debugging and getting more logs?

Comment (aggarg) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/FreeRTOSIPConfig.h:47]: Done.

**Review by tony-josi-aws — APPROVED** [review]

**Review by tony-josi-aws — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**